src/redux/cityReducer.js

```diff
@@ -0,0 +1,52 @@
+import { v4 as uuidv4 } from 'uuid';
+import locations from '../utils/locations';
+import getDetail from '../utils/fetchApi';
+
+// Defining constants for action types
+const ADD_CITIES = 'air-pollution/src/redux/ADD_CITIES';
+const FILTER_CITIES = 'air-pollution/src/redux/FILTER_CITIES';
+
+// Defining action creators
+const addCities = (payload) => ({ type: ADD_CITIES, payload });
+const filterAction = (payload) => ({ type: FILTER_CITIES, payload });
+
+// A function that fetches city data from API for each location & returns an array of city data obj
+const fetchCityData = async () => {
+  const arr = [];
+  const promises = locations.map(async (item) => {
+    const data = await getDetail(item.lat, item.lon);
+    arr.push({
+      ...data.list,
+      name: item.name,
+      id: uuidv4(),
+    });
+  });
+  await Promise.all(promises);
+  return arr;
+};
+
```

Comment:
- [ ] You did a great job 😃 on the project, but I see that you forgot to add some tests for the redux code💯, so it will be better if you add some tests for the redux code to test your code 🚀and make sure that it is work, also to follow the requirements 😊

| Requirements | 
| --- |
|   ![image](https://user-images.githubusercontent.com/83360636/235906995-162dea45-55f5-4817-a860-3d9c784fecee.png)   |

> Example: look at My redux test 😊⤵️
> If you can not see the image clearly just click on it to open a new tab in the browser😊

| My test for redux| 
 | --- |
 |   ![image](https://user-images.githubusercontent.com/83360636/235909036-628e2ac2-e288-4d1e-8b94-e0f2dc809b73.png)    |